src/util/fields.py

```diff
@@ -731,6 +733,79 @@ def convert_field_to_html(cr, model, field, skip_inherit=()):
         convert_field_to_html(cr, inh.model, field, skip_inherit=skip_inherit)
 
 
+def convert_many2one_field_to_many2many(cr, model, field, new_name=None, m2m_table=None, col1=None, col2=None):
+    """
+    Convert a Many2one field to a Many2many.
+
+    It creates the relation table and fill it.
+
+    :param str model: model name of the field to convert
+    :param str field: current name of the field to convert
+    :param str new_name: new name of the field to convert.
+                         If not provide, it's `field` with a trailing "s".
+    :param str m2m_table: name of the relation table. Automatically generated if not provided.
+    :param str col1: name of the column referencing `model`.
+    :param str col2: name of the column referencing the target of `field`.
+    """
+    _validate_model(model)
+    if new_name is None:
+        if not field.endswith("_id"):
+            raise ValueError("Please specify the new name explicitly")
+        new_name = field + "s"
+
+    table1 = table_of_model(cr, model)
+    table2, _, _ = target_of(cr, table1, field)
+
+    if m2m_table is None:
+        m2m_table = "{}_{}_rel".format(*sorted([table1, table2]))
+
+    if col1 is None:
+        col1 = "{}_id".format(table1)
+    if col2 is None:
+        col2 = "{}_id".format(table2)
+
+    create_m2m(cr, m2m_table, table1, table2, col1, col2)
+
+    dedup = SQLStr("ON CONFLICT DO NOTHING")
+    if cr._cnx.server_version < 90500:
```

Comment:
We probably never going to need this (minimal PG for valid targets it's been 10 for a while) but I guess it doesn't hurt.